net/tcp/tcp_wrbuffer.c

```diff
@@ -133,7 +136,13 @@ FAR struct tcp_wrbuffer_s *tcp_wrbuffer_alloc(void)
    * buffer
    */
 
-  net_lockedwait_uninterruptible(&g_wrbuffer.sem);
+  ret = timeout == 0 ? nxsem_trywait(&g_wrbuffer.sem) :
```

Comment:
I think https://github.com/apache/incubator-nuttx/pull/5526 was pointed out that in case of `timeout == 0` there is a trywait handling inside the `net_timedwait_uninterruptible`, so we do not need to make a selective call.

Reply:
Done